lib/msf/base/sessions/empire.rb

```diff
@@ -0,0 +1,215 @@
+      'rename_agent'=> 'Rename the curret agent name for easier interaction',
+      'agentinfo'   => 'Fetch basic information about the agent and host machine'
+    }
+  end
+
+  #
+  #HELP METHODS
+  #------------------
+  #
+  def cmd_help
+    help_table = Rex::Text::Table.new(
+      'Header'        => 'Empire Shell Commands',
+      'HeaderIndent'  => 4,
+      'Prefix'        => '\n',
+      'Postfix'       => '\n',
+      'Coloumns'      => ['Commands','Descriptions'],
+      'Indent'        => 6,
+      'SortIndex'     => -1
+    )
+    commands.each do |command, description|
+      help_table << [command, description]
+    end
+    print(help_table.to_s)
+  end
+
+  #Defining a help method to display command helps.
+  def get_help(usage, example, desc)
+    print_line "Usage : #{usage}"
+    print_line "Example : #{example}"
+    print_line
+    print_line "#{desc}"
+    print_line
+  end
+  #
+  #COMMAND METHODS
+  #----------------------
+  #
+  #Defining show_info command
+  def cmd_show_info(*args)
+    if args.length.zero? || args[0] == '-h' or args[0] == 'help'
+      return get_help("show_info <module_name>","show_info powershell/situational_awareness/network/powerview/find_gpo_location","Displays all the available options and description of the specified modules")
+    else
+      module_name = args[0]
+      @client_emp.info_module(module_name)
+    end
+  end
+
+  #Defining use_module command
+  def cmd_use_module(*args)
+    if args.length.zero? || args[0] == '-h' or args[0] == 'help'
+      return get_help("use_module <module_name>","use_module powershell/situational_awareness/network/powerview/find_gpo_location","Attempts to run the specified module against the current agent")
+    else
+      module_name = args[0]
+      @client_emp.exec_module(module_name, @agent_name)
+    end
+  end
+
+  #Defining search command
+  def cmd_search(*args)
+    if args.length.zero? || args[0] == '-h' or args[0] == 'help'
+      return get_help("search <key_term>","search privesc","Fetch all the modules resembling to your key term")
+    else
+      keyterm = args[0].to_s
+      @client_emp.search_module(keyterm)
+    end
+  end
+
+  #Defining show_modules command
+  def cmd_show_modules
+    @client_emp.get_modules
+  end
+
+  #Defining empire shell command
+  def cmd_shell(*args)
+    if args.length = 1
+      command = args[0]
+      puts @client_emp.exec_command(@agent_name, command)
+    elsif args.length > 1
+      command = args.join(" ")
+      puts @client_emp.exec_command(@agent_name, command)
+    elsif args.length.zero? || args[0] == '-h' or args[0] == 'help'
+      return get_help("shell <shell_command>","shell start notepad.exe","Runs a shell command in target host and stores the result in Empire Database. Please wait few moments before fetching the results, for results to be properly populated")
+    end
+  end
+
+  #Defining results command
+  def results(*args)
```

Comment:
shouldn't this be `cmd_results`?

Reply:
Yeah, changed them, you can have a look. Thanks for these though.